### PR TITLE
chore: refactor tutorial model logic

### PR DESCRIPTION
**Pull request by shubhammehra4**
## Description of the change

- Refactor move long messages into a separate file
- Unify all print & logger.info to use io.display_message
- Update class names/ file names to be more relevant
- Remove extra/commented code

## Linear Ticket

Resolves PRML-1007

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/predictions/profiles_mlcorelib/py_native/profiles_tutorial/model.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            self.logger.info("unzipping sample data...")
+            unzip_sample_data(project_path, self.logger)
+
+        profile_builder = ProfileBuilder(project_path, self.reader, self.fast_mode)
+        profile_builder.run(this)
+
+
+def get_sample_data_path():
+    """Returns the path to the data directory"""
+    return pkg_resources.resource_filename(
```
[the remark this answers is not in the document]
Comment (shubhammehra4) [reply src/predictions/profiles_mlcorelib/py_native/profiles_tutorial/model.py:62]: The recommended way to refer files included with the package is pkg_resources.resource_filename

File: src/predictions/profiles_mlcorelib/py_native/profiles_tutorial/yaml_generator.py [hunk 2]
```diff
@@ -91,7 +31,7 @@ def create_pb_project(self, entity_name, id_types, connection_name, id_graph_mod
             "id_types": [{"name": id_type} for id_type in id_types],
         }
 
-        with open(os.path.join(self.project_path, CONFIG_FILE_PATH), "w") as f:
+        with open(CONFIG_FILE_PATH, "w") as f:
```
[the remark this answers is not in the document]
Comment (shubhammehra4) [reply src/predictions/profiles_mlcorelib/py_native/profiles_tutorial/yaml_generator.py:34]: Yes, sorry for the confusion.

Explanation - pb tutorial will create a temp project having this model and we should not be creating these files in the temp project folder, instead in the path from where pb is running. Essentially my initial fix to add project_path needs to be undone

File: src/predictions/profiles_mlcorelib/py_native/profiles_tutorial/model.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (shekhar-rudder) [reply src/predictions/profiles_mlcorelib/py_native/profiles_tutorial/model.py:63]: [pkg_resources](https://setuptools.pypa.io/en/latest/pkg_resources.html) has been deprecated

[the remark this answers is not in the document]
Comment (shubhammehra4) [reply src/predictions/profiles_mlcorelib/py_native/profiles_tutorial/model.py:63]: Made the change, although `pkg_resources` has been only recently been deprecated that too in a version that we don't support
https://github.com/mu-editor/mu/issues/2485

[the remark this answers is not in the document]
Comment (shekhar-rudder) [reply src/predictions/profiles_mlcorelib/py_native/profiles_tutorial/model.py]: It has been removed in python 3.12. Deprecation was done earlier

File: src/predictions/profiles_mlcorelib/py_native/profiles_tutorial/model.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    except Exception as e:
+        raise Exception(f"An error occurred while extracting: {str(e)}")
+
+
+def get_sample_data_path():
+    """Returns the path to the data directory"""
+    try:
+        zip_file = (
+            resources.files("profiles_mlcorelib")
+            / "py_native/profiles_tutorial/sample_data.zip"
```
Comment (shekhar-rudder) [inline src/predictions/profiles_mlcorelib/py_native/profiles_tutorial/model.py:83]: Will `/` work in windows ?

Comment (shubhammehra4) [reply src/predictions/profiles_mlcorelib/py_native/profiles_tutorial/model.py:83]:
```
def joinpath(self, *descendants: StrPath) -> "Traversable":
        """
        Return Traversable resolved with any descendants applied.

        Each descendant should be a path segment relative to self
        and each may contain multiple levels separated by
        ``posixpath.sep`` (``/``).
        """
```

The actual method also uses /, will defer to that for safety

**Review by shekhar-rudder — APPROVED** [review]

**Pull request merged**